toctocbin/AndroidArchitecture/app/src/main/java/com/test/androidarchitecture/adpter/ViewpagerAdapter.kt

```diff
@@ -0,0 +1,31 @@
+package com.test.androidarchitecture.adpter
+
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentStatePagerAdapter
+import com.test.androidarchitecture.CoinFragment
+import com.test.androidarchitecture.data.MarketTitle
+
+class ViewpagerAdapter(fm: FragmentManager?) : FragmentStatePagerAdapter(fm) {
+
+
+    private var marketTitles: List<MarketTitle> = mutableListOf()
+
+    override fun getItem(position: Int): Fragment {
+        return CoinFragment.getInstance(marketTitles[position].marketSearch)
+    }
+
+    override fun getCount(): Int {
+        return marketTitles.size
+    }
+
+    override fun getPageTitle(position: Int): CharSequence? {
+        return marketTitles[position].marketTitle
+    }
+
+    fun setData(marketTitles: List<MarketTitle>){
+        this.marketTitles = marketTitles
```

Comment:
list 를 넣게 되면 외부에서 변경할 수 있게 되서 addAll() 로 받는게 좋을 것 같구요
그러면 adapter 에서 가지고 있는 marketTitles 를 final 로 쓸 수 있습니다
```suggestion
        this.marketTitles.addAll(marketTitles)
```

Reply:
51dc151

변경하였습니다.